### DISCUSSION TITLE
Unable to compute the item to item matrix from example code

### DISCUSSION THREAD

**Discussion by phileasme** [pyg-team/pytorch_geometric#7443] [Q&A] [answered]
Hello, 
I was just running the [ hetero bipartite_sage_unsup](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/hetero/bipartite_sage_unsup.py) example code.

However, I noticed in the example there is an interaction matrix we are computing (comat = mat.T @ mat), which breaks my instance.

I have two questions:
1) What hardware is expected to run this example?
I'm using 16vCPU's, 104GB RAM and a T4 GPU which I'm guessing wouldn't be used at that stage.
2) Why are we computing an item-to-item relation matrix from the user interactions if we are demonstrating a heterogeneous relation between users and items, couldn't we rely on these implicit relations solely? (It sounds like it would defeat the purpose).
I would assume the only item to item relation for this dataset in a heterogeneous setting that would make sense is if these items were placed in the same cart or to some extent purchased by the same user (not users).

Quite frankly new to using gnn's in pyg..
Any explanation or clarification would be very much appreciated to guide me in my learning.
Many thanks,

**Comment by rusty1s** [comment]
@husimplicity @HuxleyHu98 Do you have an understanding of how much memory we need to run this example? I agree that the sparse matrix multiplication here consumes a lot of memory, and we should probably fix this.

@phileasme A more light-weight example of this model is presented in https://github.com/pyg-team/pytorch_geometric/blob/master/examples/hetero/bipartite_sage.py. It's a special model that makes use of the item2item graph, it's not a requirement of PyG. I agree that you can achieve something similar with two rounds of message passing.

**Reply by phileasme** [reply]
Thank you @rusty1s for responding and for offering a lighter-weight example :)!

**Comment by husimplicity** [comment, marked as answer]
@rusty1s @phileasme As tested, it consumes RSS=193563432KB, i.e. about 185GB to run the example.

**Reply by phileasme** [reply]
Thank you for testing it and reporting back :)
